x-pack/plugins/fleet/server/services/agent_policies/bump_agent_policies_task.ts

```diff
@@ -0,0 +1,135 @@
+  TaskManagerStartContract,
+} from '@kbn/task-manager-plugin/server';
+import { v4 as uuidv4 } from 'uuid';
+import { uniq } from 'lodash';
+
+import { DEFAULT_SPACE_ID } from '@kbn/spaces-plugin/common';
+
+import { agentPolicyService, appContextService } from '..';
+import { runWithCache } from '../epm/packages/cache';
+import { getPackagePolicySavedObjectType } from '../package_policy';
+import { mapPackagePolicySavedObjectToPackagePolicy } from '../package_policies';
+import type { PackagePolicy, PackagePolicySOAttributes } from '../../types';
+import { normalizeKuery } from '../saved_object';
+import { SO_SEARCH_LIMIT } from '../../constants';
+
+const TASK_TYPE = 'fleet:bump_agent_policies';
+
+export function registerBumpAgentPoliciesTask(taskManagerSetup: TaskManagerSetupContract) {
+  taskManagerSetup.registerTaskDefinitions({
+    [TASK_TYPE]: {
+      title: 'Fleet Bump policies',
+      timeout: '5m',
+      maxAttempts: 3,
+      createTaskRunner: ({ taskInstance }: { taskInstance: ConcreteTaskInstance }) => {
+        let cancelled = false;
+        return {
+          async run() {
+            if (cancelled) {
+              throw new Error('Task has been cancelled');
+            }
+
+            await runWithCache(async () => {
+              await _updatePackagePoliciesThatNeedBump(appContextService.getLogger(), cancelled);
+            });
+          },
+          async cancel() {
+            cancelled = true;
+          },
+        };
+      },
+    },
+  });
+}
+
+async function getPackagePoliciesToBump(savedObjectType: string) {
+  const result = await appContextService
+    .getInternalUserSOClientWithoutSpaceExtension()
+    .find<PackagePolicySOAttributes>({
+      type: savedObjectType,
+      filter: normalizeKuery(savedObjectType, `${savedObjectType}.bump_agent_policy_revision:true`),
+      perPage: SO_SEARCH_LIMIT,
+      namespaces: ['*'],
+      fields: ['id', 'namespaces', 'policy_ids'],
+    });
+  return {
+    total: result.total,
+    items: result.saved_objects.map((so) =>
+      mapPackagePolicySavedObjectToPackagePolicy(so, so.namespaces)
+    ),
+  };
+}
+
+export async function _updatePackagePoliciesThatNeedBump(logger: Logger, cancelled: boolean) {
+  const savedObjectType = await getPackagePolicySavedObjectType();
+  const packagePoliciesToBump = await getPackagePoliciesToBump(savedObjectType);
+
+  logger.info(
+    `Found ${packagePoliciesToBump.total} package policies that need agent policy revision bump`
+  );
+
+  const packagePoliciesIndexedBySpace = packagePoliciesToBump.items.reduce((acc, policy) => {
+    const spaceId = policy.spaceIds?.[0] ?? DEFAULT_SPACE_ID;
+    if (!acc[spaceId]) {
+      acc[spaceId] = [];
+    }
+
+    acc[spaceId].push(policy);
+
+    return acc;
+  }, {} as { [k: string]: PackagePolicy[] });
+
+  const start = Date.now();
+
+  for (const [spaceId, packagePolicies] of Object.entries(packagePoliciesIndexedBySpace)) {
+    if (cancelled) {
```

Comment:
Here is where you'd checked `isCancelled()` vs the boolean value.  And note that this is a good sort of place to put this, in case the array being processed ends up being extremely large.